pylint/checkers/classes/class_checker.py

```diff
@@ -1775,15 +1796,15 @@ def _is_type_self_call(self, expr: nodes.NodeNG) -> bool:
         )
 
     @staticmethod
-    def _is_classmethod(func):
+    def _is_classmethod(func: nodes.FunctionDef) -> bool:
         """Check if the given *func* node is a class method."""
 
         return isinstance(func, nodes.FunctionDef) and (
             func.type == "classmethod" or func.name == "__class_getitem__"
         )
 
     @staticmethod
-    def _is_inferred_instance(expr, klass):
+    def _is_inferred_instance(expr: nodes.Name, klass: nodes.ClassDef) -> bool:
```

Comment:
`expr` can be other things I think.

Reply:
It's the ``expr`` attribute of a ``node: nodes.Attribute | nodes.AssignAttr`` so I think it's always a ``Name`` even though ``expr`` is typed ``NodeNg | None``. I'm going to be be safe and use ``NodeNg | None``